elide-core/src/test/java/com/yahoo/elide/core/PaginationLogicTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import com.yahoo.elide.core.pagination.Pagination;
+import org.glassfish.jersey.internal.util.collection.MultivaluedStringMap;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * Tests parsing the page params for json-api pagination.
+ */
+public class PaginationLogicTest {
+
+    @Test
+    public void shouldParseQueryParamsForCurrentPageAndPageSize() {
+        MultivaluedMap<String, String> queryParams = new MultivaluedStringMap();
+        queryParams.add("page[size]", "10");
+        queryParams.add("page[number]", "2");
+
+        Pagination pageData = Pagination.parseQueryParams(queryParams);
+
+        Assert.assertEquals(pageData.getOffset(), 10);
```

Comment:
http://jsonapi.org/format/#fetching-pagination. 

> For example, a page-based strategy might use query parameters such as page[number] and page[size], an offset-based strategy might use page[offset] and page[limit]

So the underlying structure in this case is offset (for start) and limit (for limit). So page=1 with limit 10, is 0-10.